delta/plugins/search/src/test/scala/ch/epfl/bluebrain/nexus/delta/plugins/search/SearchConfigUpdateFunctionSuite.scala

```diff
@@ -0,0 +1,127 @@
+import ch.epfl.bluebrain.nexus.delta.sdk.utils.Fixtures
+import ch.epfl.bluebrain.nexus.delta.sdk.views.ViewRef
+import ch.epfl.bluebrain.nexus.delta.sourcing.model.Identity.Subject
+import ch.epfl.bluebrain.nexus.delta.sourcing.model.{Identity, Label, ProjectRef}
+import ch.epfl.bluebrain.nexus.testkit.bio.BioSuite
+import ch.epfl.bluebrain.nexus.testkit.postgres.Doobie
+import io.circe.Json
+import munit.AnyFixture
+
+import java.time.Instant
+import scala.concurrent.duration.DurationInt
+
+class SearchConfigUpdateFunctionSuite extends BioSuite with CompositeViewsFixture with Fixtures with Doobie.Fixture {
+  override def munitFixtures: Seq[AnyFixture[_]] = List(doobie)
+  private lazy val xas                           = doobie()
+
+  implicit private val s: Identity.User = subject
+  implicit private val baseUri: BaseUri = BaseUri("http://localhost", Label.unsafe("v1"))
+
+  private val fetchContext = FetchContextDummy[CompositeViewRejection](
+    Map(project.ref -> project.context),
+    Set.empty[ProjectRef],
+    ProjectContextRejection
+  )
+
+  private lazy val compositeViews = CompositeViews(
+    fetchContext,
+    ResolverContextResolution(rcr),
+    alwaysValidate,
+    crypto,
+    config,
+    xas
+  )
+
+  private val viewDef = ActiveViewDef(ViewRef(projectRef, id), uuid, 1, viewValue)
+
+  private val newProjections: NonEmptySet[CompositeViewProjectionFields] =
+    NonEmptySet.of(esProjectionFields)
+  private val newSources: NonEmptySet[CompositeViewSourceFields]         =
+    NonEmptySet.of(projectFields)
+  private val newRebuildStrategy                                         =
+    Some(Interval(2.minutes))
+  private val newViewFields                                              =
+    viewFields.copy(
+      projections = newProjections,
+      sources = newSources,
+      rebuildStrategy = newRebuildStrategy
+    )
+
+  test("SearchConfigUpdater should run successfully") {
+    for {
+      views   <- compositeViews
+      // Create the view
+      created <- views.create(id, projectRef, viewFields)
+      _        = assertEquals(
+                   created.copy(value = created.value.copy(source = Json.obj())),
+                   resourceFor(id, viewValue, source = Json.obj())
+                 )
+      // Start the updater
+      updater  = SearchConfigUpdater.update(views)
+      _       <- updater(viewDef, newViewFields)
+      // Check that the update was successful
+
+    } yield ()
+  }
+
+  test("Sources, projections, and rebuild strategy should be updated") {
+    for {
+      views   <- compositeViews
+      updated <- views.fetch(id, projectRef)
+      _        = assertEquals(updated.rev, 2)
+      _        = assertEquals(updated.value.sources.map(_.toFields), newSources)
+      _        = assertEquals(updated.value.projections.map(_.toFields), newProjections)
+      _        = assertEquals(updated.value.rebuildStrategy, newRebuildStrategy)
+    } yield ()
+  }
```

Comment:
This is really just one test case. Consider using methods rather than comments to abstract some of the setup, that way it won't be so much boilerplate to read. If it's not clear what I mean I'd be happy to show you